### DISCUSSION TITLE
Using pyannote in Rust

### DISCUSSION THREAD

**Discussion by thewh1teagle** [pyannote/pyannote-audio#1714] [Q&A]
I would like to use pyannote from Rust with `whisper-rs` which uses `whisper.cpp`
I'm wondering about rewriting pyannote in `Rust` for using in [Vibe](https://github.com/thewh1teagle/vibe). How hard should it be to rewrite this? any thoughts?

Update:
created [pyannote-rs](https://github.com/thewh1teagle/pyannote-rs)

**Comment by altunenes** [comment]
I'm looking for something like that with Rust right now :-) what do you think is the best approach so far?

**Reply by thewh1teagle** [reply]
I think that using the onnx runtime is the most promising approach, but the thing is that pyannote-audio is not a simple solution. They use multiple models to get the job done, such as segmenting (finding speech timings) and then voice embedding on each of those segments.

So, it will require some work. Initially, I thought to use the segments returned from Whisper itself, but it turned out that it doesn't really split segments across speakers, so a segments (sentences) from whisper.cpp can contain multiple speakers.

Join the research in [vibe/issues/74](https://github.com/thewh1teagle/vibe/issues/74) :)

**Reply by altunenes** [reply]
thank you so much for the detailed message, the best is to continue in the Vibe!
